test/gtest/argmax.hpp

```diff
@@ -0,0 +1,182 @@
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <gtest/gtest.h>
+#include <miopen/argmax.hpp>
+#include <miopen/miopen.h>
+#include <random>
+
+struct ArgmaxTestCase
+{
+    size_t N;
+    size_t C;
+    size_t D;
+    size_t H;
+    size_t W;
+    int32_t dim;
+    friend std::ostream& operator<<(std::ostream& os, const ArgmaxTestCase& tc)
+    {
+        return os << " N:" << tc.N << " C:" << tc.C << " D:" << tc.D << " H:" << tc.H
+                  << " W:" << tc.W << " dim:" << tc.dim;
+    }
+
+    std::vector<size_t> GetInput()
+    {
+        if((N != 0) && (C != 0) && (D != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, D, H, W});
+        }
+        else if((N != 0) && (C != 0) && (H != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, H, W});
+        }
+        else if((N != 0) && (C != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, C, W});
+        }
+        else if((N != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, W});
+        }
+        else if((N != 0))
+        {
+            return std::vector<size_t>({N});
+        }
+        else
+        {
+            std::cout << "Error Input Tensor Lengths\n" << std::endl;
+            return std::vector<size_t>({0});
+        }
+    }
+};
+
+std::vector<ArgmaxTestCase> ArgmaxTestConfigs()
+{ // n c d h w dim
+    // clang-format off
+    return {
+        { 16, 21,   0, 513, 513, 1 },   //deeplabv3m
+        { 24, 21,   0, 513, 513, 1 },   //deeplabv3r
+        { 64, 21,   0, 230, 333, 1 },   //fcn_resnet_lraspp
+        { 64, 21,   0, 215, 288, 1 },
+        { 1,  21,   0, 333, 500, 1 },   //stdc
+        { 1,  21,   0, 375, 500, 1 },
+        { 15, 21,   0, 256, 256, 1 },   //unet
+        { 22, 21,   0, 256, 256, 1 },
+        { 21, 412,  0, 0,   500, 0 },
+        { 21, 333,  0, 0,   500, 0 }
+      };
+    // clang-format on
+}
+
+inline int32_t SetTensorLayout(miopen::TensorDescriptor& desc)
+{
+    std::vector<std::size_t> lens = desc.GetLengths();
+    std::vector<int32_t> int32_t_lens(lens.begin(), lens.end());
+
+    // set the strides for the tensor
+    return SetTensorNd(&desc, int32_t_lens, desc.GetType());
+}
+
+template <typename T = float>
+struct ArgmaxTest : public ::testing::TestWithParam<ArgmaxTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        argmax_config = GetParam();
+        std::mt19937 gen(0);
+        std::uniform_real_distribution<> d{-3, 3};
+        auto gen_value = [&](auto...) { return d(gen); };
+
+        dim = argmax_config.dim;
+
+        auto in_dims = argmax_config.GetInput();
+
+        input = tensor<T>{in_dims}.generate(gen_value);
+
+        std::vector<size_t> out_dims;
+
+        for(int i = 0; i < in_dims.size(); i++)
+        {
+            if(i != dim)
+            {
+                out_dims.push_back(in_dims[i]);
+            }
+        }
```

Comment:
[Recommendation]
`std::copy_if`?

Reply:
Since we have to use index here, the code becomes similarly long.
https://stackoverflow.com/questions/9480785/how-to-obtain-index-of-element-from-predicate-passed-to-some-stl-algorithm